[DOC] rework documentation (#892)
process-analytics/bpmn-visualization-js: README.md

        <a href="https://github.com/process-analytics/bpmn-visualization-js/releases">
          <img alt="GitHub release (latest by date including pre-releases)" src="https://img.shields.io/github/v/release/process-analytics/bpmn-visualization-js?color=orange&include_prereleases"> 
        </a> 

>         <a href="https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/master/examples/index.html">

Reviewer: :+1:  for the badge update
:question: shouldn't we change the screenshot as Hacktoberfest is done?

Reviewer 2: Yes, I asked me the same question the last week 🙂

Author: yes, definitely

> ## 🎮 Demo

Author: ```suggestion
## 🎮 Demo and examples 
```

> Please check [__⏩ demo live environment__](https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/master/examples/index.html). \

Reviewer: :question: switch title to `examples` for consistency with the description and other documentation elements?

Author: when I check some new lib I always ctrl+f 'demo' to quick search the preview
but it is subjective
maybe 'Demo examples'

> Please check [__⏩ demo live environment__](https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/master/examples/index.html). \

Author: ```suggestion
Please check [__⏩ live environment__](https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/master/examples/index.html). \
```

You will find there basic usage as well as detailed examples showing possible rendering customizations.

## 🎨 Features

The `bpmn-visualization` is in early development stage and is subject to changes prior to the `1.0.0` release.\
\
Already available features:
- [Supported BPMN Elements](https://process-analytics.github.io/bpmn-visualization-js/#supported-bpmn-elements).
- [Navigate through the BPMN diagram](https://process-analytics.github.io/bpmn-visualization-js/#diagram-navigation)

Planned features:

> - Display options for execution data with interactive capacities.

Reviewer: :warning: end of the sentence not consistent: put final dot or not

> - Display options for execution data with interactive capacities.

Author: ```suggestion
- Display options for execution data with interactive capacities
```

- BPMN extensions
- Library extension points

## 🌏 Browser Support

**We do our best to support recent versions of major browsers**

| <img src="https://www.google.com/chrome/static/images/chrome-logo.svg" alt="Chrome" width="18px" height="18px" /> Chrome | <img src="https://user-media-prod-cdn.itsre-sumo.mozilla.net/uploads/products/2020-04-14-08-36-13-8dda6f.png" alt="Firefox" width="18px" height="18px" /> Firefox | <img src="https://developer.apple.com/assets/elements/icons/safari/safari-96x96.png" alt="Safari" width="18px" height="18px" /> Safari | <img src="https://avatars0.githubusercontent.com/u/11354582?s=200&v=4" alt="Edge" width="18px" height="18px" /> Edge |
| :---------: | :---------: | :---------: | :---------: |
|  ✔️ |  ✔️ |  ✔️ |  ✔️ |

**Note**: Internet Explorer will never be supported. \
The library may work with the other browsers. They must at least support ES6.

## ♻️ Usage

> Library is available through [NPM](https://npmjs.org/package/bpmn-visualization) \

Reviewer: I would prefer something like https://www.npmjs.com/package/roughjs/v/4.3.1#install

> Library is available through [NPM](https://npmjs.org/package/bpmn-visualization) \

Reviewer 2: ```suggestion
The library is available through [NPM](https://npmjs.org/package/bpmn-visualization). \
```

We support various module formats such as:
[IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE),
[ESM](https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm),

> [CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs)

Author: ```suggestion
[CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs).
```

> [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE),
> [ESM](https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm),
> [CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs).

Reviewer 2: ```suggestion
[IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE): `bpmn-visualization.js`
[ESM](https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm): `bpmn-visualization.esm.js`
[CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs): `bpmn-visualization.cjs.js`
```

  
Here are examples per module: \
[bpmn-visualization (IIFE)](https://github.com/process-analytics/bpmn-visualization-examples/blob/master/examples/01-getting-started) \

> [bpmn-visualization (ESM); Typescript + Rollup](https://github.com/process-analytics/bpmn-visualization-examples/tree/master/examples/projects/typescript-vanilla-with-rollup)

Reviewer: I would remove it, why not providing few commands a most project does?
Further details are available in the examples.
We won't be able to put enough information in the README on that topic, so I suggest to drop everything instead of just putting information that are not enough for a whole understanding and the whole usage

> Here are examples per module: \
> [bpmn-visualization (IIFE)](https://github.com/process-analytics/bpmn-visualization-examples/blob/master/examples/01-getting-started) \
> [bpmn-visualization (ESM); Typescript + Rollup](https://github.com/process-analytics/bpmn-visualization-examples/tree/master/examples/projects/typescript-vanilla-with-rollup)

Author: ```suggestion
```

⚒️ If you want to configure yourself:
* Install the dependency in your **package.json** file:
```shell script
npm i bpmn-visualization
```

> * Load necessary scripts (replace _X._Y._Z by the recent version)

Author: ```suggestion
* Load bpmn-visualization script (replace _X._Y._Z by the recent version)
```

> * Load bpmn-visualization script (replace _X._Y._Z by the recent version)

Reviewer: :question: can we use `{version}` instead of x.y.z?

> * Load bpmn-visualization script (replace _X._Y._Z by the recent version)

Author: ```suggestion
* Load bpmn-visualization script (replace {version} by the recent version)
```

[IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE),
[ESM](https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm),
[CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs).
  

### IIFE
See how it's done in [this example](https://github.com/process-analytics/bpmn-visualization-examples/blob/master/examples/01-getting-started).

### ESM
🚀 You can find an integration example of Vanilla Typescript with rollup, [here](https://github.com/process-analytics/bpmn-visualization-examples/tree/master/examples/projects/typescript-vanilla-with-rollup).
Here are examples per module: \
[bpmn-visualization (IIFE)](https://github.com/process-analytics/bpmn-visualization-examples/blob/master/examples/01-getting-started) \
[bpmn-visualization (ESM); Typescript + Rollup](https://github.com/process-analytics/bpmn-visualization-examples/tree/master/examples/projects/typescript-vanilla-with-rollup)

⚒️ If you want to configure yourself:
* Install the dependency in your **package.json** file:
```shell script
npm i bpmn-visualization
```

* Load necessary scripts 
* Load bpmn-visualization script (replace _X._Y._Z by the recent version)
```html

>     <!-- load global settings -->

Reviewer: nice, I hated this!

>     <script src="https://unpkg.com/bpmn-visualization@_X._Y._Z/dist/bpmn-visualization.js"></script>

Author: ```suggestion
    <script src="https://unpkg.com/bpmn-visualization@{version}/dist/bpmn-visualization.js"></script>
```

>       return  ``; // your BPMN 2.0 XML content

Reviewer: :warning: we shouldn't suggest to return an empty content
can we replace it by `.....`? otherwise let's drop this part

Reviewer 2: Why introduce a function?

I find that it complicates the reading of the doc.

The user can define the content of his diagram as he wishes afterwards. Maybe, we can add a sentence for that.

>     function bpmnDiagram() {
>       return  ``; // your BPMN 2.0 XML content
>     }

Author: ```suggestion
    let bpmnContent; // your BPMN 2.0 XML content
```

> * Define the mxGraph container

Author: ```suggestion
* Define the bpmn diagram container
```

> * Define the bpmn diagram container

Reviewer: ```suggestion
* In the HTML page, define the container that displays the BPMN diagram
```

Reviewer: I don't like switching from js to html then js back. Not fluid and messy

Reviewer 2: Me too :)

> * Initialize BpmnVisualization from the container and load the diagram

Author: ```suggestion
* Initialize BpmnVisualization and load the diagram
```

>     const bpmnVisualization = new bpmnvisu.BpmnVisualization(document.getElementById('bpmn-container'));

Reviewer 2: ```suggestion
    const bpmnContainerElt = document.getElementById('bpmn-container');
    const bpmnVisualization = new bpmnvisu.BpmnVisualization(bpmnContainerElt);
```

>     bpmnVisualization.load(bpmnDiagram());

Reviewer 2: As I said before I prefer to let as it was in order to have the code the more simple as possible.
If the user want more complexity, he can see the examples.

```suggestion
    bpmnVisualization.load(bpmnContent);
```

**Note**: Internet Explorer will never be supported. \
The library may work with the other browsers. They must at least support ES6.

## ♻️ Usage
The library is available from [NPM](https://npmjs.org/package/bpmn-visualization). \
We support various module formats such as:
- [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE): `dist/bpmn-visualization.js`
- [ESM](https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm): `dist/bpmn-visualization.esm.js`
- [CommonJS](https://www.typescriptlang.org/docs/handbook/2/modules.html#commonjs): `dist/bpmn-visualization.cjs.js`
  

⚒️ If you want to configure yourself:
* Install the dependency in your **package.json** file:
```shell script
npm i bpmn-visualization
```

> * Load bpmn-visualization script (replace {version} by the recent version)

Author: ```suggestion
* In the HTML page:
   * bpmn-visualization script (replace {version} by the recent version)
   * define the container that displays the BPMN diagram. Here _bpmn-container_
```

```html

>     <script src="https://unpkg.com/bpmn-visualization@{version}/dist/bpmn-visualization.js"></script>

Author: ```suggestion
    <script src="https://unpkg.com/bpmn-visualization@{version}/dist/bpmn-visualization.js"></script>
    ...
    <div id="bpmn-container"></div>
```
